Internal Terminology sever not working in github actions

>Caused by: org.hl7.fhir.utilities.SimpleHTTPClient$HTTPResultException: Invalid HTTP response 404 from http://localhost:8082/matchboxv3/tx-reg/resolve?fhirVersion=4.0&url=urn%3Aietf%3Abcp%3A47&usage=validation&nocache=1712698406607 (null) (Response in /tmp/http-log/fhir-http-2.log)

the validation call seems to work from the client to the matchbox server, however the terminology call from within the matchbox server itself seems to fail

https://github.com/ahdis/matchbox/actions/runs/8614498842

https://productionresultssa4.blob.core.windows.net/actions-results/53b24acd-3b7b-4f44-bd43-10baf89c2bf4/workflow-job-run-c892508d-390a-5765-0ad0-b341c7fd0dd4/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-04-10T06%3A45%3A15Z&sig=E6r1NtfiA4ygmTOk6cw4fMIDhpDskzbxPCKZy8G8xQ8%3D&sp=r&spr=https&sr=b&st=2024-04-10T06%3A35%3A10Z&sv=2021-12-02

The official terminology server also responds with an error 404 for that call: https://tx.fhir.org/r4/tx-reg/resolve?_format=application/fhir+json
It seems to be still unimplemented there.

The issue is a bad lock in Lucene files:
```
2024-04-09T21:33:52.4350234Z 2024-04-09 21:33:52.392 [Hibernate Search - default backend - Worker thread - 0] WARN  o.h.s.b.l.l.w.i.IndexWriterProvider [IndexWriterProvider.java:92] HSEARCH000052: An index writer operation failed. Resetting the index writer and forcing release of locks. Context: index 'ResourceTable'

2024-04-09T21:33:54.4195196Z org.springframework.orm.jpa.JpaSystemException: Unable to perform afterTransactionCompletion callback: HSEARCH800024: Unable to index documents for indexing after transaction completion: HSEARCH700124: Indexing failure: HSEARCH600016: Unable to index entity of type 'ResourceTable' with identifier '1' and tenant identifier 'null': Lock held by this virtual machine: /home/runner/work/matchbox/matchbox/matchbox-int-tests/target/lucenefiles/ResourceTable/write.lock

2024-04-09T21:33:54.9094145Z Caused by: org.hibernate.search.util.common.SearchException: HSEARCH600016: Unable to index entity of type 'ResourceTable' with identifier '1' and tenant identifier 'null': Lock held by this virtual machine: /home/runner/work/matchbox/matchbox/matchbox-int-tests/target/lucenefiles/ResourceTable/write.lock
2024-04-09T21:33:54.9096953Z Context: index 'ResourceTable'
2024-04-09T21:33:54.9097982Z 	at org.hibernate.search.backend.lucene.work.impl.AddEntryWork.execute(AddEntryWork.java:36)
2024-04-09T21:33:54.9099586Z 	at org.hibernate.search.backend.lucene.work.impl.AddEntryWork.execute(AddEntryWork.java:17)
2024-04-09T21:33:54.9101606Z 	at org.hibernate.search.backend.lucene.orchestration.impl.LuceneBatchedWorkProcessor.submit(LuceneBatchedWorkProcessor.java:63)
2024-04-09T21:33:54.9103997Z 	at org.hibernate.search.backend.lucene.orchestration.impl.LuceneBatchedWork.submitTo(LuceneBatchedWork.java:26)
2024-04-09T21:33:54.9106063Z 	at org.hibernate.search.backend.lucene.orchestration.impl.LuceneBatchedWork.submitTo(LuceneBatchedWork.java:14)
2024-04-09T21:33:54.9108092Z 	at org.hibernate.search.engine.backend.orchestration.spi.BatchingExecutor$BatchWorker.work(BatchingExecutor.java:189)
2024-04-09T21:33:54.9110063Z 	at org.hibernate.search.engine.backend.orchestration.spi.SingletonTask$RunnableWrapper.run(SingletonTask.java:199)
2024-04-09T21:33:54.9111669Z 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
2024-04-09T21:33:54.9112889Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
2024-04-09T21:33:54.9114505Z 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
2024-04-09T21:33:54.9116556Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
2024-04-09T21:33:54.9118064Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
2024-04-09T21:33:54.9119259Z 	at java.base/java.lang.Thread.run(Thread.java:1583)
```

--- a/src/test/java/health/matchbox/server/IgValidateR4Test.java
+++ b/src/test/java/health/matchbox/server/IgValidateR4Test.java
@@ -25,6 +25,7 @@ import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
 import org.springframework.context.ApplicationContext;
+import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.ContextConfiguration;
 import org.yaml.snakeyaml.Yaml;
@@ -56,6 +57,7 @@ import static org.junit.jupiter.api.Assertions.assertEquals;
 @ContextConfiguration(classes = {Application.class})
 @ActiveProfiles("validate-r4")
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@DirtiesContext
 public class IgValidateR4Test {
 
 	private static final String TARGET_SERVER = "http://localhost:8082/matchboxv3/fhir";

--- a/src/test/java/health/matchbox/server/IgValidateRawProfileTest.java
+++ b/src/test/java/health/matchbox/server/IgValidateRawProfileTest.java
@@ -18,6 +18,7 @@ import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
 import org.springframework.core.io.ClassPathResource;
 import org.springframework.core.io.Resource;
+import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.ContextConfiguration;
 
@@ -36,6 +37,7 @@ import static org.junit.jupiter.api.Assertions.assertEquals;
 @ContextConfiguration(classes = {Application.class})
 @ActiveProfiles("validate-raw-profile")
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@DirtiesContext
 public class IgValidateRawProfileTest {
 
 

